src/ant/plus/heartrate.py

```diff
@@ -29,61 +29,27 @@
 
 from __future__ import print_function
 
-from threading import Lock
+from .plus import DeviceProfile
 
-from ant.plus.plus import _EventHandler
 
+class HeartRate(DeviceProfile):
+    """ANT+ Heart Rate Monitor"""
 
-class HeartRateCallback(object):
-    """Receives heart rate events.
-    """
+    channelPeriod = 8070
+    deviceType = 0x78
+    name = 'Heart Rate'
 
-    def device_found(self, device_number, transmission_type):
-        """Called when a device is first detected.
-
-        The callback receives the device number and transmission type.
-        When instantiating the HeartRate class, these can be supplied
-        in the device_id and transmission_type keyword parameters to
-        pair with the specific device.
+    def __init__(self, node, network, callbacks=None):
         """
-        pass
-
-    def heartrate_data(self, computed_heartrate, event_time_ms, rr_interval_ms):
-        """Called when heart rate data is received.
-
-        computed_heartrate - computed heart rate from monitor
-        event_time_ms - event time in ms
-        rr_interval_ms - time between successive R waves
+        :param node: The ANT node to use
+        :param network: The ANT network to connect on
+        :param callbacks: Dictionary of string-function pairs specifying the callbacks to
+                use for each event. In addition to the events supported by `DeviceProfile`,
+                `BicyclePower` also has the following:
```

Comment:
Seems like BicyclePower should be HeartRate here.

Reply:
Yes, your right. Damn copy paste!